core/src/main/java/org/ehcache/core/internal/util/Pacer.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.core.internal.util;
+
+import org.ehcache.core.spi.time.TimeSource;
+
+/**
+ * Class used to pace a call to prevent it from happening too frequently.
+ */
+public class Pacer {
+
+  private volatile long nextLogTime = 0;
```

Comment:
Right now we're going to get a flurry of calls whenever the threshold is reached if we're multithreaded.  Should this be an AtomicLong?

Reply:
Also... I think this should be initialized to `TimeSource.getTimeMillis()` otherwise we will be broken with impl's that return negative timestamps.

Reply:
Ok for negative timestamps.

Reply:
I'm not sure the AtomicLong is that necessary since it doesn't matter that much to have many calls when a delay expires but I will pull out a stricter version. Tell me what you think.